src/components/tags/Tag.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react'
+
+import { ButtonColor } from '../../styles/button-theme'
+import { fontSize } from '../../styles/theme'
+import { SecondaryButton } from '../buttons/SecondaryButton'
+import { ArrayParam, useQueryParams, withDefault, encodeQueryParams, QueryParamConfig } from 'use-query-params';
+import { urls } from '../../domain/website';
+import { navigate } from '@reach/router';
+import { stringify } from 'query-string';
+
+interface ITag {
+  label?: string;
+  active?: boolean
+}
+
+interface IQuery {
+  tags: string[],
+}
+
+const Tag: React.FC<ITag> = ({ label, active }) => {
+  const [query] = useQueryParams<IQuery>({ tags: withDefault(ArrayParam, []) });
+
+  const onClick = () => {
+    if (label) {
+      let tags = query.tags.slice()
+      const isTag = tags.find((tag: string) => tag === label);
+      if (!isTag) {
+        tags.push(label)
+      } else {
+        tags = tags.filter((tag: string) => tag !== label)
+      }
+      const encodedQuery = encodeQueryParams({ tags: withDefault(ArrayParam, []) }, { tags });
```

Comment:
Ten fragment dobrze byłoby przesunąć do osobnego pliku opisującego logikę (może się to nazwyać {cośtam-cośtam}-service.ts
Łatwiej potem połapać się w tym i łatwiej do tego napisać osobny test bez sprawdzania w ogóle samego komponentu.
Tutaj potrzebujesz po prostu wydzielonej funkcji (dobrze nazwanej), która jak rozumiem dla konkretnego label zwraca odpowiedni URL, którego potem używasz w funkcji navigate.